dmoj/cptbox/utils.py

```diff
@@ -1,11 +1,62 @@
+import errno
 import io
+import mmap
+import os
+from tempfile import NamedTemporaryFile
+from typing import Optional
 
 from dmoj.cptbox._cptbox import memory_fd_create, memory_fd_seal
+from dmoj.cptbox.tracer import FREEBSD
 
 
 class MemoryIO(io.FileIO):
-    def __init__(self) -> None:
-        super().__init__(memory_fd_create(), 'r+')
+    _name: Optional[str] = None
+
+    def __init__(self, prefill: Optional[bytes] = None, seal=False) -> None:
+        if FREEBSD:
```

Comment:
I think we either want to branch on some sort of "expected size" here, or figure out something to make the memory backing this file be more likely to be swapped out.

If the input is 5 GiB large, we don't want to buffer all of it in memory.

Reply:
Leaving this up to the caller for now. If the caller desires, it can use `NamedFileIO`.